cumulusci/robotframework/SalesforcePlaywright.py

```diff
@@ -126,14 +133,83 @@ class 'slds-template__container', but a different locator can
             if locator is None
             else locator
         )
-        self.browser.get_element(locator)
+        self.browser.get_elements(locator)
         self.browser.execute_javascript(function=WAIT_FOR_AURA_SCRIPT)
 
         # this seems to fail once in a while for an unknown reason, so we'll
         # try twice. This seems to be more effective than calling the
-        # function once and waiting 30 seconds.
+        # function once for a longer timeout
         try:
-            self.browser.wait_until_network_is_idle("15 seconds")
+            self.browser.wait_until_network_is_idle(timeout)
         except Exception as e:
-            self.builtin.log(f"caught error waiting for idle: {e}", "WARN")
-            self.browser.wait_until_network_is_idle("15 seconds")
+            self.builtin.log(f"caught error waiting for idle: {e}", "DEBUG")
+            self.browser.wait_until_network_is_idle(timeout)
+
+    @capture_screenshot_on_error
+    def wait_until_salesforce_is_ready(
+        self, login_url, locator=None, timeout="30 seconds"
+    ):
+        """Attempt to wait until we land on a lightning page
+
+        In addition to waiting for a lightning page, this keyword will
+        also attempt to wait until there are no more pending ajax
+        requests.
+
+        The timeout parameter is taken as a rough guideline. This
+        keyword will actually wait for half of the timeout before
+        starting checks for edge cases.
+
+        """
+
+        timeout_seconds = timestr_to_secs(timeout)
+        start_time = time.time()
+
+        locator = locator if locator else "div.slds-template__container"
```

Comment:
Could also use the `or` here to keep things even more concise
```suggestion
        locator = locator or "div.slds-template__container"
```